### ISSUE TITLE
Can't enable "federation relay"

### ISSUE THREAD

**Issue by 33b5e5** [tootsuite/mastodon#8485]
I'm the admin of raki.social. I upgraded to 2.5.0rc1 from the release tag. 

The upgrade went fine, but I haven't been able to get the new "federation relay" feature working. After adding the suggested relay (https://relay.joinmastodon.org/inbox) it shows up "Disabled":

![screen shot 2018-08-27 at 11 12 04 am](https://user-images.githubusercontent.com/2781306/44677163-1c7a0400-a9ea-11e8-82b2-6c8579c0b2a7.png)

If I click "Enable", it seems like it's trying to do something, but nothing really happens -- there's no error and it remains "Disabled".

I don't know how to troubleshoot this further. I'm not sure where to look for relevant logs. I tried grepping through syslog but didn't see anything related.

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

**Comment by Marlinc** [comment]
I ran into the same issue

**Comment by ThibG** [comment]
It shows as `disabled` if the relay hasn't replied back to your follow request. Which it probably didn't do yet because it's lagging behind.
The UI should probably show that it's pending, rather than disabled!

**Comment by 33b5e5** [comment]
Ah... okay @ThibG. I will wait for a while and see if it gets the reply. Thanks.

I agree the UI should be improved. Displaying "Pending" would be a great start.

**Comment by Gargron** [comment]
I think it's a mistake on my part... As @ThibG mentions I forgot to change the UI from enabled/disabled to disabled/pending/enabled, and there might be a bug with actually recording the "accept" event because the relay seems to be sending them (and is not lagging behind significantly) and yet this issue exists.

**Comment by carstenwawer** [comment]
Got the same problem. Should we wait for the next release?

**Comment by noellabo** [comment]
Problems may occur when upper case is included in inbox_url.

It may be fixed by adding the following modification to pub-relay.
https://source.joinmastodon.org/mastodon/pub-relay/merge_requests/2

**Comment by mdbraber** [comment]
@noellabo have you tried this change? I've just recompiled my binary on relay.mastodon.nl to test but I can't see any change.

**Comment by Gargron** [comment]
Indeed, it seems that it hasn't fixed this problem.

There are test specs for handling the Accept activities in Mastodon, and the tests pass. And the relay code looks like it's sending them. So it's very mysterious. The payloads don't look very different between test spec and relay either.

**Comment by noellabo** [comment]
@mdbraber What is the error recorded in the worker's log?

```
Aug 30 07:38:52 hrt worker[22827]: POST https://dtp-mstdn.jp/inbox 401
```

**Comment by mdbraber** [comment]
I've been messing around with this and found a solution (I'm using relay.mastodon.nl as my test server). The only visible change I could see in the payloads for the pub-relay server was that it's using multiple contexts:

`"@context": {"https://www.w3.org/ns/activitystreams", "https://w3id.org/security/v1"},`

when changing this to:

`"@context": {"https://www.w3.org/ns/activitystreams"},`

in `inbox_handler.cr` and `pub_relay.cr` my relay is shown as enabled.

I'm not sure though if changing the pub-relay code should be the proposed solution or that something should be fixed on the side of Mastodon. @Gargron can probably best judge that.

**Comment by 33b5e5** [comment]
I was able to add https://relay.mastodon.nl/inbox as a relay. 👍

**Comment by u1-liquid** [comment]
I also got the same problem and I can say https://relay.mastodon.nl/inbox is working well.

**Comment by mdbraber** [comment]
FYI for all those using relay.mastodon.nl I've now created a whitelist with only selected instances to not have non-Dutch related content pushed to it. My idea is to have it act as a "booster" for Dutch related content, not a a general relay. As soon as I get round to it I'd be happy to set up another test instance. But I think @Gargron (or someone who controls relay.joinmastodon.org) might be able to fix that too.

**Comment by gled-rs** [comment]
https://relay.mastodon.host/inbox works too if needed.

**Comment by Gargron** [comment]
Are you telling me that all other installations of the relay work and only mine doesn't? You're also using the Crystal code, right?

**Comment by gled-rs** [comment]
Seems to be the case :)On Aug 30, 2018 5:28 PM, Eugen Rochko <notifications@github.com> wrote:Are you telling me that all other installations of the relay work and only mine doesn't? You're also using the Crystal code, right?

**Comment by Gargron** [comment]
That was a fantastic pointer actually. I started suspecting it's something about my deployment of the code, and lo and behold, a high `-c` setting has led to a lot of delivery jobs in the worker failing because of too many open files.

This is still something to be solved because there are *a lot* of jobs and the relay must keep up with them, but at least the mystery is solved, I lowered the setting and was able to enable a relay on mastodon.social.
